libp2pdht/private/eth/p2p/discoveryv5/node.nim

```diff
@@ -99,9 +99,9 @@ func `==`*(a, b: Node): bool =
 func hash*(id: NodeId): Hash =
   hash(id.toByteArrayBE)
 
-proc random*(T: type NodeId, rng: var BrHmacDrbgContext): T =
+proc random*(T: type NodeId, rng: var HmacDrbgContext): T =
   var id: NodeId
-  brHmacDrbgGenerate(addr rng, addr id, csize_t(sizeof(id)))
+  hmacDrbgGenerate(rng, addr id, csize_t(sizeof(id)))
```

Comment:
@Menduist re: [node.nim#L104](https://github.com/status-im/nim-libp2p-dht/blob/5a04a699754aaa1d9e041af99e543cf1b534e811/libp2pdht/private/eth/p2p/discoveryv5/node.nim#L104) with these changes:
```diff
-proc random*(T: type NodeId, rng: var BrHmacDrbgContext): T =
+proc random*(T: type NodeId, rng: var HmacDrbgContext): T =
   var id: NodeId
-  brHmacDrbgGenerate(addr rng, addr id, csize_t(sizeof(id)))
+  hmacDrbgGenerate(rng, addr id, csize_t(sizeof(id)))
```
That shouldn't run afoul of the problem you [described](https://github.com/status-im/nim-codex/pull/248#issuecomment-1246396945) in my nim-codex PR, because [this impl](https://github.com/status-im/nim-bearssl/blob/f4c4233de453cb7eac0ce3f3ffad6496295f83ab/bearssl/abi/bearssl_rand.nim#L43-L44) of `proc hmacDrbgGenerate` has the first parameter as `var HmacDrbgContext` as does `proc random` here, is that correct? Just double-checking my understanding.

Reply:
Yes, here it's the caller responsability to do
```nim
NodeId.random(rng[])
# and not
let rng = rng[]
NodeId.random(rng)
```

Reply:
(issue with `var` parameter is that they don't work in async function, which is why other librairies take a `ref HmacDrbgContext` instead of a `var`)